presto-resource-group-managers/src/main/java/com/facebook/presto/resourceGroups/ResourceEstimatesRule.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.resourceGroups;
+
+import com.facebook.presto.spi.session.ResourceEstimates;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.airlift.json.JsonCodec;
+import io.airlift.units.DataSize;
+import io.airlift.units.Duration;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import static io.airlift.json.JsonCodec.jsonCodec;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public final class ResourceEstimatesRule
```

Comment:
What do you think about getting rid of the `rule` suffix from the class/method/variable names?

Reply:
@nezihyigitbasi : Let's call it range.